robot_sf/data_analysis/plot_dataset.py

```diff
@@ -0,0 +1,186 @@
+        #     break
+
+    for timestep, vels in enumerate(velocity_list):
+        plt.scatter([timestep] * len(vels), vels, alpha=0.5, c="blue", s=1)
+
+    plt.xlabel("Time Step")
+    plt.ylabel("Velocity")
+    plt.title("Pedestrian Velocity over Time, raw" if raw else "Pedestrian Velocity over Time")
+    # plt.legend()
+    plt.tight_layout()
+
+    title = "all_npc_ped_velocities_raw" if raw else "all_npc_ped_velocities"
+    if unique_id:
+        filename = f"robot_sf/data_analysis/plots/{title}_{unique_id}.png"
+    else:
+        filename = f"robot_sf/data_analysis/plots/{title}.png"
+    plt.savefig(filename)
+    if interactive:
+        plt.show()
+    plt.close()
+
+
+def plot_ego_ped_acceleration(
+    ego_ped_acceleration: list, interactive: bool = False, unique_id: str = None
+):
+    """
+    Plot the acceleration of the ego pedestrian.
+
+    Args:
+        ego_ped_acceleration (list): List of ego pedestrian accelerations.
+        interactive (bool): If True, show the plot interactively.
+        unique_id (str): Unique identifier for the plot filename, usually the timestamp
+    """
+
+    plt.plot(ego_ped_acceleration, label="Acceleration")
+    plt.xlabel("Timestep")
+    plt.ylabel("Acceleration")
+    plt.title("Ego Ped Acceleration over Time")
+    plt.legend()
+    if unique_id:
+        filename = f"robot_sf/data_analysis/plots/ego_ped_acc_{unique_id}.png"
+    else:
+        filename = "robot_sf/data_analysis/plots/ego_ped_acc.png"
+    plt.savefig(filename)
+    if interactive:
+        plt.show()
+    plt.close()
+
+
+def plot_ego_ped_velocity(
+    ego_ped_acceleration: list, interactive: bool = False, unique_id: str = None
+):
+    """
+    Plot the velocity of the ego pedestrian based on the acceleration.
+
+    Args:
+        ego_ped_acceleration (list): List of ego pedestrian accelerations.
+        interactive (bool): If True, show the plot interactively.
+        unique_id (str): Unique identifier for the plot filename, usually the timestamp
+    """
+
+    ego_ped_velocity = []
+    cumulative_sum = 0.0
+    for acc in ego_ped_acceleration:
+        cumulative_sum += acc
+        # Clip, because the ego pedestrian can't go faster than 3 m/s
+        cumulative_sum = np.clip(cumulative_sum, None, 3)
+        ego_ped_velocity.append(cumulative_sum)
+
+    ego_ped_velocity = np.array(ego_ped_velocity)
+
+    plt.plot(ego_ped_velocity, label="Velocity")
+    plt.xlabel("Timestep")
+    plt.ylabel("Velocity")
+    plt.title("Ego Ped Velocity over Time")
+    plt.legend()
+
+    if unique_id:
+        filename = f"robot_sf/data_analysis/plots/ego_ped_vel_{unique_id}.png"
+    else:
+        filename = "robot_sf/data_analysis/plots/ego_ped_vel.png"
+    plt.savefig(filename)
+    if interactive:
+        plt.show()
+    plt.close()
+
+
+def main():
+    # filename = "robot_sf/data_analysis/datasets/2025-02-06_10-24-12.json"
+    filename = "robot_sf/data_analysis/datasets/2025-01-16_11-47-44.json"
```

Comment:
Here, you explicitly use a a json file that is not commited to the repository. Please design your code in a way that it runs everywhere. For example, add a json file for testing purposes to the testing folder and use this file as your "default" arguments. @JuliusMiller